python/lsst/meas/astrom/anetAstrometry.py

```diff
@@ -0,0 +1,464 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import numpy
+from contextlib import contextmanager
+
+import lsst.pex.exceptions
+import lsst.afw.geom as afwGeom
+from lsst.afw.cameraGeom import TAN_PIXELS
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from .astrom import Astrometry
+from lsst.meas.astrom.sip import makeCreateWcsWithSip
+from lsst.afw.table import Point2DKey, CovarianceMatrix2fKey
+
+class ANetAstrometryConfig(pexConfig.Config):
+    solver = pexConfig.ConfigField(
+        dtype = Astrometry.ConfigClass,
+        doc = "Configuration for the astrometry solver",
+    )
+    forceKnownWcs = pexConfig.Field(dtype=bool, doc=(
+        "Assume that the input image's WCS is correct, without comparing it to any external reality." +
+        " (In contrast to using Astrometry.net).  NOTE, if you set this, you probably also want to" +
+        " un-set 'solver.calculateSip'; otherwise we'll still try to find a TAN-SIP WCS starting " +
+        " from the existing WCS"), default=False)
+    rejectThresh = pexConfig.RangeField(dtype=float, default=3.0, doc="Rejection threshold for Wcs fitting",
+                                        min=0.0, inclusiveMin=False)
+    rejectIter = pexConfig.RangeField(dtype=int, default=3, doc="Rejection iterations for Wcs fitting",
+                                      min=0)
+
+
+    ## \addtogroup LSST_task_documentation
+    ## \{
+    ## \page ANetAstrometryTask
+    ## \ref ANetAstrometryTask_ "ANetAstrometryTask"
+    ## \copybrief ANetAstrometryTask
+    ## \}
+
+class ANetAstrometryTask(pipeBase.Task):
+    """!
+    \anchor ANetAstrometryTask_
+
+    \brief Use astrometry.net to match input sources with a reference catalog and solve for the Wcs
+
+    \warning This task is currently in \link lsst.pipe.tasks.astrometry\endlink;
+    it should be moved to lsst.meas.astrom.
+
+    The actual matching and solving is done by the 'solver'; this Task
+    serves as a wrapper for taking into account the known optical distortion.
+
+    \section pipe_tasks_astrometry_Contents Contents
+
+     - \ref pipe_tasks_astrometry_Purpose
+     - \ref pipe_tasks_astrometry_Initialize
+     - \ref pipe_tasks_astrometry_IO
+     - \ref pipe_tasks_astrometry_Config
+     - \ref pipe_tasks_astrometry_Debug
+     - \ref pipe_tasks_astrometry_Example
+
+    \section pipe_tasks_astrometry_Purpose	Description
+
+    \copybrief ANetAstrometryTask
+
+    Calculate an Exposure's zero-point given a set of flux measurements of stars matched to an input catalogue.
+    The type of flux to use is specified by ANetAstrometryConfig.fluxField.
```

Comment:
This looks like documentation for a photometric calibration Task.  That's not what this does, right?